libs/common/domain/src/lib/model/record/metadata.model.ts

```diff
@@ -226,7 +226,16 @@ export interface DatasetRecord extends BaseRecord {
   temporalExtents: Array<DatasetTemporalExtent>
   spatialRepresentation?: SpatialRepresentationType
 }
+
 export type DatasetFeatureCatalog = {
+  featureTypes: Array<{
+    name: string
+    definition: string
+    featureAttributes: DatasetFeatureAttributes
```

Comment:
I think this property should be called `attributes` and only contain an array, instead of another object with a single property